frb_codegen/src/generator/dart/mod.rs

```diff
@@ -514,6 +540,22 @@ fn generate_wire2api_func(
     )
 }
 
+fn generate_opaque_func(ty: &IrType) -> Acc<String> {
+    Acc {
+        io: format!(
+            "late final ffi.NativeFinalizer _{0}Finalizer = ffi.NativeFinalizer(inner._drop_opaque_{0}Ptr);
+            ffi.NativeFinalizer get {0}Finalizer => _{0}Finalizer;",
+            ty.dart_api_type(),
+        ),
+        wasm: format!(
+            "late final Finalizer<int>  _{0}Finalizer = Finalizer(inner.drop_opaque_{0});
+            Finalizer<int> get {0}Finalizer => _{0}Finalizer;",
+            ty.dart_api_type(),
+        ),
+        ..Default::default()
+    }
+}
+
```

Comment:
would be great to merge the two branches since they only have minor differences.

Using the `typedef` trick, maybe `ffi.NativeFinalizer` and `Finalizer<int>` can be typedef as `GeneralFinalizer` etc.